pkg/build/generator/generator.go

```diff
@@ -432,6 +432,17 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 	}
 
 	build.Name = getNextBuildName(bc)
+	// If there is a naming conflict, continue to increase the version until we
+	// get a unique name. If that fails, keep the +10 version increase without
+	// reporting an error. When the Build is created, it will fail with a naming
+	// error
+	if _, err := g.Client.GetBuild(ctx, build.Name); err == nil {
```

Comment:
Is this the right level to be iterating until we find a free name? Are all callers intending to generate a new build even if one already exists?